x-pack/plugins/cloud_security_posture/common/runtime_mappings/get_safe_posture_type_runtime_mapping.ts

```diff
@@ -16,14 +16,16 @@ export const getSafePostureTypeRuntimeMapping = (): MappingRuntimeFields => ({
     type: 'keyword',
     script: {
       source: `
-        def postureTypeAvailable = doc.containsKey("rule.benchmark.posture_type") &&
-          !doc["rule.benchmark.posture_type"].empty;
+        def postureTypeAvailable = doc.containsKey("rule.benchmark.posture_type") && !doc["rule.benchmark.posture_type"].empty;
+        boolean isNativeCsp = doc.containsKey("data_stream.dataset") && doc["data_stream.dataset"].value == "cloud_security_posture.findings";
```

Comment:
these painless scripts are very brittle, let's add `&& !doc["data_stream.dataset"].empty` as well to the check

Reply:
sure, i removed this check because we pretty much know that data_stream.dataset always exist but i can add it back.

Regarding your comment, removing this whole process right now is risky, a lot of pieces are relying on it, not worth taking the risk atm.